lib/src/DOtherSide.cpp

```diff
@@ -135,12 +137,53 @@ ::DosQQmlContext *dos_qqmlapplicationengine_context(::DosQQmlApplicationEngine *
     return engine->rootContext();
 }
 
+void dos_qqmlapplicationengine_addimageprovider(DosQQmlApplicationEngine *vptr, DosQQuickImageProvider *vptr_i)
+{
+  auto engine = static_cast<QQmlApplicationEngine *>(vptr);
+  auto provider = static_cast<QQuickImageProvider *>(vptr_i);
+  engine->addImageProvider(QString("test"), provider);
+}
+
 void dos_qqmlapplicationengine_delete(::DosQQmlApplicationEngine *vptr)
 {
     auto engine = static_cast<QQmlApplicationEngine *>(vptr);
     delete engine;
 }
 
+
+::DosQQuickImageProvider *dos_qquickimageprovider_create()
+{
+    return new DosImageProvider();
+}
+
+void dos_qquickimageprovider_delete(::DosQQuickImageProvider *vptr)
+{
+    auto provider = static_cast<QQuickImageProvider *>(vptr);
```

Comment:
See my previous comment on the create function. This case your casting from void* to QQuickImageProvider* thus in the create method you should return QQuickImageProvider* and not a DosImageProvider*